torch/lib/c10d/reducer.cpp

```diff
@@ -137,6 +137,8 @@ Reducer::Reducer(
           gradAccToVariableMap_[grad_accumulator.get()] = index;
         }
 
+        numGradHooksTriggeredMap_[index] = 0;
```

Comment:
I am adding some additional maps in https://github.com/pytorch/pytorch/pull/55075 as well. I wonder if we should eventually make some sort of struct to hold all of these so that they are easy to update all together and not scattered all over the place in `reducer.cpp`.

Reply:
possibly we can do later on when more semantically group-able fields are added